packages/cli/fern-definition/validator/src/rules/compatible-ir-versions/compatible-ir-versions.ts

```diff
@@ -0,0 +1,95 @@
+                    if (!cliRelease.ok) {
+                        return [];
+                    }
+                    const cliIrVersion = cliRelease.body.irVersion;
+
+                    // Pull the generator release to get the IR version
+                    let invocationIrVersion: number;
+                    if (invocation["ir-version"] != null) {
+                        // You've overridden the IR version in the generator invocation, let's clean it up
+                        invocationIrVersion = Number(invocation["ir-version"].replace("v", ""));
+                        if (cliIrVersion >= invocationIrVersion) {
+                            // You've overridden the IR version and you're on a CLI that works with it!
+                            return [];
+                        }
+                    } else {
+                        const generatorEntity = await fdr.generators.getGeneratorByImage({
+                            dockerImage: invocation.name
+                        });
+                        if (!generatorEntity.ok) {
+                            return [];
+                        }
+                        const generatorRelease = await fdr.generators.versions.getGeneratorRelease(
+                            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                            generatorEntity.body!.id,
+                            invocation.version
+                        );
+
+                        if (generatorRelease.ok) {
+                            // We've pulled the generator release, let's get it's IR version
+                            invocationIrVersion = generatorRelease.body.irVersion;
+                            if (cliIrVersion >= generatorRelease.body.irVersion) {
+                                // You're on a CLI that works with the generator you have specified!
+                                return [];
+                            }
+                        } else {
+                            // We don't have a generator release for this generator, so we can't validate it.
+                            return [];
+                        }
+                    }
+
+                    // If we've made it this far, we know the IR versions aren't a match, let's grab the min version
+                    // and throw an error for the user telling them what to upgrade to to use this generator.
+                    const minCliVersion = await fdr.generators.cli.getMinCliForIr(invocationIrVersion);
+                    if (minCliVersion.ok) {
+                        return getMaybeBadVersionMessage(invocation.name, minCliVersion.body.version, cliVersion) ?? [];
+                    } else {
+                        throw new Error(
+                            `Failed to get min CLI version for IR version ${invocationIrVersion} ${JSON.stringify(
+                                minCliVersion
+                            )}`
+                        );
```

Comment:
why are we throwing here? ideally we shouldnt be failing if the network request fails